pkg/monitor/cluster/nsgreconciliations.go

```diff
@@ -0,0 +1,53 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	subnetscontroller "github.com/Azure/ARO-RP/pkg/operator/controllers/subnets"
+)
+
+func (mon *Monitor) emitNSGReconciliation(ctx context.Context) error {
+	co, err := mon.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	reconciled := 0
+	updated, err := annotationUpdated(co.Annotations)
+	if err != nil {
+		return err
+	}
+
+	if updated {
+		reconciled = 1
+	}
+
+	mon.emitGauge("nsg.reconciliations", int64(reconciled), nil)
+
+	return nil
+}
+
+func annotationUpdated(annotations map[string]string) (bool, error) {
+	if annotations == nil {
+		return false, nil
+	}
+
+	t := annotations[subnetscontroller.AnnotationTimestamp]
+	timestamp, err := time.Parse(time.RFC1123, t)
+	if err != nil {
+		return false, err
+	}
+
+	if time.Since(timestamp) < time.Second*120 {
```

Comment:
I think metrics are emitted not in a fixed interval, but as quickly as it takes to run the stack once. Meaning with a fixed lookback time of 120 seconds, we would read the same event multiple times, and even worse, that  behavior would change if we add more metrics, or if the stack takes different times to run, depending on some cluster metric. Could we maybe cut off seconds and only take full minutes?

Reply:
We have a ticker inside the `func (mon *monitor) worker(stop <-chan struct{}, delay time.Duration, id string)` function, so we actually emit metrics every 60 seconds